# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1331831007 14400
#      Thu Mar 15 13:03:27 2012 -0400
# Branch stable
# Node ID 21ac4b576003cfee0d571d257a4c7f64bec408b0
# Parent  af552038cc520311ab60aba66c2bb27dd3a5a856
fix scalar indexing in anonymous function bug (bug #34765)

* ov-base-scalar.h (octave_base_scalar<T>::do_multi_index_op):
New function.

diff --git a/src/ov-base-scalar.h b/src/ov-base-scalar.h
--- a/src/ov-base-scalar.h
+++ b/src/ov-base-scalar.h
@@ -66,16 +66,19 @@ public:
   octave_value_list subsref (const std::string& type,
                              const std::list<octave_value_list>& idx, int)
     { return subsref (type, idx); }
 
   octave_value subsasgn (const std::string& type,
                          const std::list<octave_value_list>& idx,
                          const octave_value& rhs);
 
+  octave_value_list do_multi_index_op (int, const octave_value_list& idx)
+    { return do_index_op (idx); }
+
   bool is_constant (void) const { return true; }
 
   bool is_defined (void) const { return true; }
 
   dim_vector dims (void) const { static dim_vector dv (1, 1); return dv; }
 
   octave_idx_type numel (void) const { return 1; }
 
